src/cargo/sources/registry/index.rs

```diff
@@ -52,6 +52,13 @@ impl<'cfg> RegistryIndex<'cfg> {
         if self.cache.contains_key(name) {
             return Ok(self.cache.get(name).unwrap());
         }
+        // If the lock file doesn't already exist then this'll cause *someone*
+        // to create it. We don't actually care who creates it, and if it's
+        // already there this should have no effect.
+        drop(OpenOptions::new()
+                        .write(true)
+                        .create(true)
+                        .open(self.path.join(INDEX_LOCK).into_path_unlocked()));
```

Comment:
Hm, this is an index *lock*. Is this lock happening in place in the local index? That is, is it writing to the local registry's directory tree? Seems like cargo should not assume that the local registry is writable.